src/Feature/Theme/GlobalColors.re

```diff
@@ -587,7 +587,8 @@ module Oni = {
 
   module Sneak = {
     let background =
-      define("oni.sneak.background", all(ref(Selection.background)));
+      define("oni.sneak.background", all(ref(Menu.selectionBackground)
+      ));
```

Comment:
Looks like this changed in https://github.com/onivim/oni2/pull/1635 - I moved it back to what it was prior. Is there something I should look out in changing this back, @glennsl ?

One thing we could consider is setting just a legible default, and customizing our built-in themes with oni.sneak colors.

Reply:
Hmm, no I don't think so. When I changed this I think my assumption was that `selection.background` was more general and would be used as fallback for the more specific selection colors, so that lazy theme authors could get away with just defining that and not bother with the more specific ones. But that assumption didn't hold, and I knew that when I merged the PR, but I missed this one it seems.

> One thing we could consider is setting just a legible default

Or use a single theme color and derive the other by computing the opposite, for example, as both we and vscode do for the cursor.

Reply:
Cool, thanks for the details.

> Or use a single theme color and derive the other by computing the opposite, for example, as both we and vscode do for the cursor.

Hmm, didn't think about that, that's a good idea!

I tried it with each color - unfortunately there were still a few themes that were less than optimal, 

ie: `Selection.background` + opposite for foreground with gruvbox:
![image](https://user-images.githubusercontent.com/13532591/80645651-4639c880-8a20-11ea-9836-c5e9b2f0df78.png)

`Menu.selectionBackground` + opposite with laserwave:
![image](https://user-images.githubusercontent.com/13532591/80645801-7ed9a200-8a20-11ea-9370-0aa93a557696.png)
(not as bad, but a bit hard to see on my lower contrast monitor..)

But it's good to know we have that in our toolbox.

Reply:
Hmm, yeah it would be nice to have a smarter one that maximizes contrast. `opposite`, which think I made emulate vscode's behaviour, just does `1 - intensity` for each of the color components, which obviously won't work very well for values close to 0.5.

Reply:
That makes sense - I wonder if another color space would work better for finding the opposite (like [HSL](https://github.com/onivim/oni2/pull/1687). Some interesting ideas here too: https://gamedev.stackexchange.com/questions/38536/given-a-rgb-color-x-how-to-find-the-most-contrasting-color-y

Reply:
If the color is converted to HSL it would be pretty easy I think. The "opposite" is just `hue + 0.5 mod 1.0`. The saturation can probably remain untouched, and the luminance is mostly what determines the contrast, so just adjust that up or down to until the difference is sufficient (but not maximized, as that would just be black or white)

Reply:
> I wonder if another color space would work better for finding the opposite (like HSL

Ah, great minds think alike!